packages/knip/src/typescript/resolveModuleNames.ts

```diff
@@ -35,6 +34,32 @@ export function createCustomModuleResolver(
   const customCompilerExtensionsSet = new Set(customCompilerExtensions);
   const extensions = [...DEFAULT_EXTENSIONS, ...customCompilerExtensions];
 
+  const virtualDeclarationFiles = new Map<string, { path: string; ext: string }>();
+
+  const tsSys: ts.System = {
```

Comment:
Unless there's a good reason not to, I think we could move `virtualDeclarationFiles` and `tsSys` out of the function scope, easier for moving/refactors later.

Reply:
My thinking behind this was that `virtualDeclarationFiles` is specific to the instance of `resolveModuleName` that is created by `createCustomModuleResolver`. I think it is theoretically possible for two instances with different compiler settings to interfere with each other. Also, keeping `virtualDeclarationFiles`  local and not global allows it to be garbage collected. Whether all this is important in practice, I don’t know. But conceptually I feel like stateful things should be localized.

If you agree, I’ll keep it there, otherwise I can move it out.

Reply:
You're right!